### PR TITLE
Upgrade job should upgrade from 1.20

### LINKED ISSUE #1555: Upgrade job should upgrade from 1.20
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-periodic-capi-e2e-main is broken since https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1524 merged because it's trying to upgrade from 1.19 to 1.21 which is not allowed. We should change the `KUBERNETES_VERSION_UPGRADE_FROM` variable default 
in https://github.com/nader-ziada/cluster-api-provider-azure/blob/08f456963fb47f801e75eae730837e3cffe57bfc/test/e2e/config/azure-dev.yaml to `v1.20.9` which is the latest 1.20 patch.

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Rahul-D78**
**What type of PR is this?**

 /kind bug

**What this PR does / why we need it**:

We should change the KUBERNETES_VERSION_UPGRADE_FROM variable default
in https://github.com/nader-ziada/cluster-api-provider-azure/blob/08f456963fb47f801e75eae730837e3cffe57bfc/test/e2e/config/azure-dev.yaml to v1.20.9 which is the latest 1.20 patch.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #1555 , ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1555 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/ok-to-test
/test pull-cluster-api-provider-azure-capi-e2e

testing the upgrade job

**Comment by CecileRobertMichon** [comment]
@Rahul-D78 please link the issue this fixes in the PR description next to "Fixes #"

**Comment by CecileRobertMichon** [comment]
flake

/retest
/lgtm

**Comment by Rahul-D78** [comment]
HI @CecileRobertMichon I have added to issue number is it going to work now

**Comment by CecileRobertMichon** [comment]
the upgrade jobs are still failing, need to investigate what's going on

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -103,7 +103,7 @@ variables:
   ETCD_VERSION_UPGRADE_TO: "3.4.3-0"
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:103]: @Rahul-D78 can you please update etcd and coredns versions to:

```
ETCD_VERSION_UPGRADE_TO: "3.4.13-0"
COREDNS_VERSION_UPGRADE_TO: "1.8.4"
```

Comment (Rahul-D78) [reply test/e2e/config/azure-dev.yaml:103]: yes, sure

**Comment by Rahul-D78** [comment]
I have made the desired changes I hope It passes all test cases

**Comment by CecileRobertMichon** [comment]
/retest

just checking if we got unlucky...

**Comment by CecileRobertMichon** [comment]
I was able to repro locally. New upgraded nodes are not becoming ready, investigating.

```
NAME                                   STATUS     ROLES                  AGE     VERSION
default-template-control-plane-8dhn9   Ready      control-plane,master   17m     v1.20.9
default-template-md-0-hbl5n            Ready      <none>                 15m     v1.20.9
default-template-md-0-qlrqc            NotReady   <none>                 6m49s   v1.21.2
default-template-md-0-vlvnd            Ready      <none>                 14m     v1.20.9
```

/hold

**Comment by Rahul-D78** [comment]
yes thanks for investing .

**Comment by CecileRobertMichon** [comment]
/close

After investigation (#1568) and per https://github.com/kubernetes-sigs/cluster-api/issues/4896, the failing tests are to be deprecated. Closing this PR in favor of #1566 which will remove these tests.

**Pull request closed without merging**